test/e2e/ingress.go

```diff
@@ -243,31 +374,68 @@ var _ = Describe("GCE L7 LoadBalancer Controller", func() {
 		framework.beforeEach()
 		client = framework.Client
 		ns = framework.Namespace.Name
-		Expect(waitForRCPodsRunning(client, "kube-system", "glbc")).NotTo(HaveOccurred())
-		Expect(checkLeakedResources()).NotTo(HaveOccurred())
+		// Scaled down the existing Ingress controller so it doesn't interfere with the test.
+		Expect(scaleKubeSystemRC(client, lbContainerName, 0)).NotTo(HaveOccurred())
```

Comment:
I agree it's not-ideal, however it's the exact same pod. I guess you're recommending that we somhow pass a uid into a template and apply it to the cluster addon pod definition. That's possible, but unprecedented (i.e all the addons are in cluster/addons and not cluster/salt). 

A lot of this confusion comes from the fact that we don't know how to coordinate in the ubernetes context yet, or rather, with multiple kubernetes masters. I think this will take care of itself when we solve that problem, but I'm welcome to ideas.